CMakeLists.txt

```diff
@@ -0,0 +1,100 @@
+#New CMake build file for FireHorn
+cmake_minimum_required(VERSION 3.25)
+
+project(flight_computer VERSION 0.1 LANGUAGES C CXX)
+
+# Set the build type to Debug
+set(CMAKE_BUILD_TYPE Debug)
+
+include_directories(include)
+
+# Define the source files
+set(SOURCES 
+    src/flightControl/AvState.cpp
+    tests/test_AvState.cpp
+)
+
+# Create the executable
+add_executable(flight_control_test ${SOURCES})
+
+add_library(ADXL375 src/sensor/adxl375.cpp include/sensor/adxl375.h)
+target_include_directories(ADXL375 PUBLIC include/sensor)
+target_link_libraries(ADXL375 PRIVATE I2CInterface)
+
+add_library(BMP3 src/sensor/bmp3.cpp include/sensor/bmp3.h)
+target_include_directories(BMP3 PUBLIC include/sensor)
+target_link_libraries(BMP3 PRIVATE I2CInterface)
+
+add_library(BMI08x src/sensor/bmi08xa.cpp src/sensor/bmi08a.cpp src/sensor/bmi08g.cpp include/sensor/bmi08x.h include/sensor/bmi08.h)
+target_include_directories(BMI08x PUBLIC include/sensor)
+target_link_libraries(BMI08x PRIVATE I2CInterface)
+
+add_library(I2CInterface src/I2CInterface/I2CInterface.cpp include/I2CInterface/I2CInterface.h include/I2CInterface/i2c_wrappers.h)
+target_include_directories(I2CInterface PUBLIC include/I2CInterface)
+target_link_libraries(I2CInterface PRIVATE pigpio)
+
+add_library(I2CGPS src/sensor/I2CGPS.cpp src/sensor/TinyGPS++.cpp include/sensor/I2CGPS.h include/sensor/TinyGPS++.h)
+target_include_directories(I2CGPS PUBLIC include/sensor)
+target_link_libraries(I2CGPS PRIVATE I2CInterface)
```

Comment:
interfaces should be subdirectories to be able to mock them and unit test them more efficiently